common/go-utils/src/inv.rs

```diff
@@ -165,7 +168,10 @@ impl TryFrom<&GoFile> for Artifact {
             semantic_version: Version::parse_go(&value.version)?,
             os: value.os.parse::<Os>()?,
             arch: value.arch.parse::<Arch>()?,
-            sha_checksum: value.sha256.clone(),
+            checksum: Checksum {
```

Comment:
The length won't be validated here since you aren't parsing `value.sha256`, right?

Reply:
Good point! I've added a [`Checksum::new` constructor](https://github.com/heroku/buildpacks-go/pull/234/commits/10d0b84cdee7aa434e25537f05d998140445c3e9) that enforces length validation, and made [`Checksum#algorithm` private](https://github.com/heroku/buildpacks-go/pull/234/commits/e93ffce64dc3959875b86433949a8bd1ae3a3fc3) to force callers to use the new constructor. I think this should address this concern?